api/src/main/java/com/cloudera/livy/LivyClientBuilder.java

```diff
@@ -57,6 +57,13 @@ public LivyClientBuilder setAll(Properties props) {
     return this;
   }
 
+  public LivyClientBuilder setIfMissing(String key, String value) {
```

Comment:
Not sure how likely it is that two threads would call this method at the same time, we might want to `synchronize` this.